RFCXXX-Native-Json-Adapters.md

```diff
@@ -0,0 +1,258 @@
+. . .
+```
+
+### Follow-on Work
+
+A number of follow-on activities should be considered to bring continued improvement of the native command experience.
+
+#### Improved TabCompletion
+
+Current tab-completion (on Unix) is done via a somewhat fragile call to the native shell.
+Tab-completion on Windows must be implemented specifically.
+we should take the opportunity to improve our tooling around tab-completion to make it easier to create,
+and provide generalized help parsers (where possible) to generate tab-completion.
+`bash` provides a number of more generalized tab-completion, we should attempt to do the same for PowerShell users.
+
+For example, bash has a single completer all known hosts which is used by nearly 20 executables.
+We already will tab-complete for file system items,
+a small effort should be able to increase our coverage for other native executables.
+
+#### Improved Formatting Tooling
+
+Once text output is converted to objects,
+we should improve the process for creating formatting directives.
+When creating a formatting file,
+the user needs to manually author the `XML` and then call `update-formatdata` to add that configuration to the current session.
+Usually, formatting files are associated with a module and distributed in this way.
+With stand-alone native applications, it means that providing for reasonable formatting will require an additional file.
+A more dynamic approach for formatting would improve the experience for the user.
+
+## Open Questions
+
+### History
+
+It's not clear whether history should be altered to show the added pipeline elements. We don't currently do that for `out-default` which can also add to the pipeline.
+It should be relatively easy to determine whether the pipeline as the `app-json`, as `get-command app` will include the name of the json adapter.
+
+### Adaptation of Native Apps by Fully Qualified Path
+
+Should the automatic json adaptation be suppressed when a fully qualified path (`/bin/ls` vs `ls`) is provided?
+This may be problematic as most scripted solutions may construct a full path to the application,
+and relying on `$env:PATH` could be used as a code injection attack vector.
+The expectation is that the `<app>-json` converter will be found as either a function, script, or other native app.
+If a script, or other native app we would use normal command discovery to find the converter,
+which would not be specified with fully qualified path.
+Regardless of mechanism, it should be a feature to temporarily disable automatic adaptation.
+
+### Selective non-adaptation
+
+Should there be a way to suppress adaptation on a single invocation.
+Shells such `bash` have a way to set a variable for a single invocation:
+
+```bash
+$ P=. ls -ld $P
+drwxr-xr-x  10 james  staff  320 Dec  6 12:05 .
+```
+
+Can or should something like this be done for PowerShell?
+
+## Alternatives
```

Comment:
There is 2.5 year old proposal which cover all scenarios from the RFC and even more
https://github.com/PowerShell/PowerShell/issues/13428
And as it was said there, it's great in combination with https://github.com/dotnet/command-line-api